### ISSUE TITLE
about classifier baseline

### ISSUE THREAD

**Issue by whySnowwW** [cyvius96/few-shot-meta-baseline#12]
Hello, I wonder if the classifier baseline branch can be adapted in a few shot learning task?

**Comment by cyvius96** [comment]
Hi. I didn't get the point maybe. Classifier-baseline is trained with cross entropy loss, after training it can perform few-shot learning tasks by cosine nearest-centroid.

**Comment by whySnowwW** [comment]
Thanks for your help. Do you mean that the classifier baseline can do some beautiful work when I use it in a few shot classification task?

**Comment by whySnowwW** [comment]
And I also have some trouble in getting the output of the prediction of the classifier. Is that the logit ?

**Comment by cyvius96** [comment]
Yes, the variable 'logits' in the code refers to the predicted logits of current batch.

**Comment by whySnowwW** [comment]
And i also get some question on the classifier baseline:  
i wonder if the model use the encoder to solve the few shot classification issue by cosine distance in the code or through the meta learning method?
and also , i find that in the test-few-shot.py, the dataloader label is the index of the true label, am i correct or is that some kind of mistake i made?

**Comment by cyvius96** [comment]
Classifier baseline does not perform meta-learning (more details are in the paper).
In the code, the labels are "local label", i.e. the label in current N-way K-shot task.

**Comment by whySnowwW** [comment]
If so,  does the code perform few shot classifier in the classifier baseline through using cosine distance?

**Comment by whySnowwW** [comment]
And in the train_classifier.py , there is a few-shot eval , what is this stage used for?

**Comment by cyvius96** [comment]
Yes, it classifies with cos distance. In train_classifier.py few-shot eval is evaluating the few-shot performance of current model (i.e. applying cos nearest-centroid to current classifier for few-shot tasks). It is for checking the trend of the performance.

**Comment by whySnowwW** [comment]
Thanks for your patience, but i still have an issue why do we need test dataset in the train stage?

**Comment by cyvius96** [comment]
You're welcome.
It is evaluating the model after each epoch, so that we can see a performance curve during training (which is common in most machine learning models' training).

**Comment by cyvius96** [comment]
You're welcome.
It is evaluating the model after each epoch, so that we can see a performance curve during training. You can delete this stage if you are not interest in these information.
